Diagnostics/HealthChecker/DataCollection/ExchangeInformation/IISInformation/Get-IISWebSite.ps1

```diff
@@ -42,20 +42,87 @@ function Get-IISWebSite {
 
         $webConfigExists = Test-Path $configurationFilePath
         $webConfigContent = $null
+        $webConfigContentXml = $null
         $validWebConfig = $false
+        $customHeaderHstsObj = [PSCustomObject]@{
+            enabled             = $false
+            "max-age"           = 0
+            includeSubDomains   = $false
+            preload             = $false
+            redirectHttpToHttps = $false
+        }
+        $customHeaderHsts = $null
 
         if ($webConfigExists) {
             $webConfigContent = (Get-Content $configurationFilePath -Raw).Trim()
 
             try {
-                [xml]$webConfigContent | Out-Null
+                $webConfigContentXml = [xml]$webConfigContent
                 $validWebConfig = $true
             } catch {
                 # Inside of Invoke-Command, can't use Invoke-CatchActions
                 Write-Verbose "Failed to convert IIS web config '$configurationFilePath' to xml. Exception: $($_.Exception)"
             }
         }
 
+        if ($validWebConfig) {
+            <#
+                HSTS configuration can be done in different ways:
+                Via native HSTS control that comes with IIS 10.0 Version 1709.
+                See: https://learn.microsoft.com/iis/get-started/whats-new-in-iis-10-version-1709/iis-10-version-1709-hsts
+                The native control stores the HSTS configuration attributes in the <hsts> element which can be found under each <site> element.
+                These settings are returned when running the Get-WebSite cmdlet and there is no need to prepare the data as they are ready for use.
+
+                Via customHeader configuration (when running IIS older than the version mentioned before where the native HSTS config is not available
+                or when admins prefer to do it via customHeader as there is no requirement to do it via native HSTS control instead of using customHeader).
+                HSTS via customHeader configuration are stored under httpProtocol.customHeaders element. As we get the content in the previous
+                call, we can simply use these data to extract the customHeader with name Strict-Transport-Security (if exists) and can then prepare
+                the data for further processing.
+                The following code searches for a customHeader with name Strict-Transport-Security. If we find the header, we then extract the directive
+                and return them as PSCustomObject. We're looking for the following directive: max-age, includeSubDomains, preload, redirectHttpToHttps
```

Comment:
Is it possible to configure this and have it work on the `web.config` file for the virtual directory? We can have `customHeaders` section in that file as well. 

Is it possible to have `customHeaders` on the `applicationHost.config` ?

Reply:
We might need to update or docs on this, as i am guessing this could be very bad: 
![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/4af07ad4-046d-45fd-bf6f-44cfae19e773)

![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/678238f0-9313-45d6-b35e-99b6a263732c)

Trying to make sure that `applicationHost.config` got updated. Found out we created a new configuration file. 

![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/617e9b98-affd-4c61-9b9f-1c6f5de84500)

![image](https://github.com/microsoft/CSS-Exchange/assets/22776718/1ddaebca-6e10-4fd2-bd12-8a4ce8fa2c42)

Reply:
Actually, this should be that bad. We should still inherit the settings from the original `applicationHost.config` file as well.

Reply:
Do you think we should add more context on this to the HSTS documentation on learn.microsoft.com?
This is what we have right now:
<img width="595" alt="image" src="https://github.com/microsoft/CSS-Exchange/assets/40993616/1f19e323-00df-4fde-88db-2d564b37abac">

Reply:
Not sure, we do state **not** to do it, but we know how good some people are at reading.

Reply:
Yeah, let's see how this goes and if we get escalations regarding this. We could then change it to a red warning box or improve the wording.